arch/arm/core/cortex_m/mpu/arm_mpu_v7_internal.h

```diff
@@ -106,6 +106,17 @@ static inline void _get_mpu_ram_region_attr(arm_mpu_region_attr_t *p_attr,
 	p_attr->rasr = _get_region_attr(1, ap, 1, 1, 1, 1, 0, size);
 }
 
+static inline void _get_mpu_nocache_region_attr(arm_mpu_region_attr_t *p_attr,
+	u32_t ap, u32_t base, u32_t size)
+{
+	/* in ARMv7-M MPU the base address is not required
+	 * to determine region attributes.
+	 */
+	(void) base;
+
+	p_attr->rasr = _get_region_attr(1, ap, 0, 0, 0, 1, 0, size);
```

Comment:
These attributes implies strongly ordered access to the region. Is that needed for regular memory?
IMHO normal, non-cacheable region (TEX=001, C=0, B=0) will give better performance (but you will have to use barriers to ensure that all memory access completed).

Reply:
I agree the attributes should be relaxed. Your suggestion looks good to me, I'll update that in the next version.